protocol-specs/generators/dpos_random_seed_generation/index.js

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { hash } = require('@liskhq/lisk-cryptography');
+
+const BaseGenerator = require('../base_generator');
+const { list: sampleDelegateList } = require('./forger_list');
+
+/* eslint-disable new-cap, no-continue */
+
+const activeDelegates = 101;
+const standByDelegates = 2;
+
+const strippedHash = data => {
+	let input;
+
+	if (typeof data === 'number') {
+		input = Buffer.from(data.toString(16), 'hex');
+	} else if (typeof data === 'string') {
+		input = Buffer.from(data, 'hex');
+	} else if (data instanceof Buffer) {
+		input = data;
+	} else {
+		throw new Error('Hash input is not a valid type');
+	}
+
+	return hash(input)
+		.slice(0, 16)
+		.toString('hex');
+};
+
+const bitwiseXOR = hexArray => {
+	if (hexArray.length === 1) {
+		return hexArray[0];
+	}
+
+	const buffers = hexArray.map(hexStr => Buffer.from(hexStr, 'hex'));
+	const bufferSizes = new Set(buffers.map(buffer => buffer.length));
+	if (bufferSizes.size > 1) {
+		throw new Error('All input for XOR should be same size');
+	}
+	const outputSize = [...bufferSizes][0];
+	const result = Buffer.alloc(outputSize, 0, 'hex');
```

Comment:
do we need to initialize with 0 as hex? I think it's sufficient to do alloc (it will be initialize with 0)

Reply:
Having an explicit init value is safer than considering an implicit default value.